### PR TITLE
Move EF and data to own layer

### LINKED ISSUE #115: Move EF stuff into its own project
The context, models (just `Cfp`) and migrations should be moved to a separate project to enable building a new API leveraging existing code.

### PR DESCRIPTION
**Pull request by jfversluis**
Moves everything that has to do with data into it' own project/layer.

I guess it could be abstracted a bit more, but that's for the next iteration

Fixes #115 
cc: @rickvdbosch

### CODE REVIEW

File: CfpExchange.Common/CfpExchange.Common.csproj [hunk 1]
```diff
@@ -7,6 +7,7 @@
   <ItemGroup>
     <PackageReference Include="HtmlAgilityPack" Version="1.11.24" />
     <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
+    <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="3.1.5" />
```
Comment (rickvdbosch) [inline CfpExchange.Common/CfpExchange.Common.csproj:10]: Is this package necessary in this location for this change?

**Comment by rickvdbosch** [comment]
Ah... I approved twice bit still didn't see the check turn green.
I don't have write access 🤦🏻‍♂️

**Comment by jfversluis** [comment]
The change is _that_ good

**Comment by jfversluis** [comment]
Besides the build errors that is

**Pull request merged**